utils/collection/parseLists.go

```diff
@@ -58,18 +58,26 @@ func ParseCommaSeparatedList(input string) []string {
 
 // ParseCommaSeparatedListToMap returns a map of key value pairs from a string containing a comma separated list
 func ParseCommaSeparatedListToMap(input string) (pairs map[string]string, err error) {
-	inputSplit := ParseCommaSeparatedList(input)
-	numElements := len(inputSplit)
+	pairs, err = ConvertSliceToMap[string](ParseCommaSeparatedList(input))
+	return
+}
+
+// ConvertSliceToMap converts a slice of elements into a map e.g. [key1, value1, key2, values2] -> {key1: value1, key2: value2}
+func ConvertSliceToMap[T comparable](input []T) (pairs map[T]T, err error) {
```

Comment:
```suggestion
func ConvertSliceToMap[T any](input []T) (pairs map[T]T, err error) {
```
I don't think we need the restriction here as well

Reply:
we do because to be a key in a map, it has to be comparable

Reply:
got it thanks for explaining